src/blocks.zig

```diff
@@ -229,9 +229,25 @@ pub fn getTypeById(id: []const u8) u16 {
 	}
 }
 
-pub fn parseBlock(data: []const u8) Block {
+pub fn parseBlock(id: []const u8) Block {
+	if(parseBlock2(id)) |parsedBlock| {
+		var block: Block = .{.typ = parsedBlock.typ, .data = 0};
+		if(parsedBlock.data) |blockData| {
+			block.data = blockData;
+		} else {
+			block.data = block.mode().naturalStandard;
+		}
+		return block;
+	} else {
+		std.log.err("Couldn't find block {s}. Replacing it with air...", .{id});
+		return .{.typ = 0, .data = 0};
+	}
+}
+
+pub fn parseBlock2(data: []const u8) ?struct {typ: u16, data: ?u16} {
```

Comment:
I need more information about block expression I have parsed, since block with no data specifier and block with data specifier have different valid meanings - block with no data specifier means match that block with any data, while block with data specifier means match that block with exact data match. I decided that since I don't know what is the preferred way to implement this I will just go with simple-stupid way and see what comes up in review.

Reply:
I would prefer if there is just one global function to parse a block. 
So maybe for now just check if it contains two colons at the call site, and maybe in the future (since this change would be too big) we can maybe return an extra enum parameter that specifies whether it contained the data or not.

Reply:
`parseBlock` is used in ~20 places and current contract is not acceptable for many reasons. It completely supresses any parsing errors, but adding same error handling to every of those ~20 places I waste of code and time IMO, hence it was made a adapter function. This way parsing is done indeed with one function, but it is `parseBlock2`, while `parseBlock` is just an adapter with error handling. Maybe we should change name of `parseBlock2` to `tryParseBlock`? We discussed something similar before, if I recall correctly in network rewrite for `receive`.

Reply:
Alright, how about this:
1. `parseBlock`, which behaves like it does now.
2. `getBlockById` which returns only a `?u16` for the block type, and ignores the data part
3. `getBlockData` which returns only a `?u16` for the data, and ignores the block id

Reply:
Do I use `getBlockById` and `getBlockData` to implement `parseBlock` or do I leave it with current implementation?